mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala

```diff
@@ -124,10 +126,18 @@ private[feature] trait StringIndexerBase extends Params with HasHandleInvalid wi
 
     val outputFields = inputColNames.zip(outputColNames).flatMap {
       case (inputColName, outputColName) =>
-        schema.fieldNames.contains(inputColName) match {
-          case true => Some(validateAndTransformField(schema, inputColName, outputColName))
-          case false if skipNonExistsCol => None
-          case _ => throw new SparkException(s"Input column $inputColName does not exist.")
+        try {
+          val dtype = _transformDataset.col(inputColName).expr.dataType
```

Comment:
Yeah this LTGM. No need to add `_` in the variable name tho.

Reply:
renamed .